providers/implementations/kdfs/argon2.c

```diff
@@ -1024,33 +1023,34 @@ static int kdf_argon2_derive(void *vctx, unsigned char *out, size_t outlen,
     if (!ossl_prov_is_running() || !kdf_argon2_set_ctx_params(vctx, params))
         return 0;
 
-    ctx->mac = EVP_MAC_fetch(ctx->libctx, "blake2bmac", ctx->propq);
+    if (ctx->mac == NULL)
+        ctx->mac = EVP_MAC_fetch(ctx->libctx, "blake2bmac", ctx->propq);
```

Comment:
We're not considering the possibility that a new provider might be loaded into the libctx and that a different implementation gets found?

The overhead for fetching this and the digest isn't significant IMO.  Argon2 is designed to be slow and consume resources.

Reply:
This is actually very arguable whether just loading a new provider should suddenly mean changes of implementation to be used within an existing kdf ctx. I mean the argon2 kdf implementation won't be changed in such case, so why the underlying mac and hash should be.